### PR TITLE
Skip uninitialized outputs during accuracy minification

### LINKED ISSUE #93437: Accuracy minifier can find spurious accuracy failures involving uninitialized memory

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #186564

Accuracy minification can truncate an FX graph suffix and use an intermediate
node as the candidate graph output. When that intermediate node is an
uninitialized tensor factory such as aten.new_empty or prims.empty_strided,
the candidate compares undefined allocation contents. On CUDA, stale caching
allocator contents can make that candidate appear to be an accuracy failure
even though the original graph initialized the tensor with a later write.

Add a minifier hook that lets callers reject suffix-output candidates, and wire
accuracy minifier entry points to skip direct outputs from uninitialized tensor
factory ops. Compile/runtime minification continues to use the existing generic
search behavior. The skipped targets cover ATen empty/new_empty variants and
PrimTorch empty factories that can appear after decomposition.

Fixes #93437
Generated by my agent

Test Plan:
- python test/functorch/test_minifier.py -k test_skip_uninitialized_suffix_outputs
- python test/functorch/test_minifier.py -k test_skip_prims_empty_strided_suffix_outputs
- python test/functorch/test_minifier.py
- python test/inductor/test_minifier.py -k test_after_aot_cpu_accuracy_error
- python test/dynamo/test_minifier.py -k test_after_dynamo_accuracy_error_cpu
- python - <<'PY'
  import torch._dynamo.repro.aoti
  print('aoti import ok')
  PY
- git diff --check
- lintrunner -a

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says the two Dr. CI failures failed during third-party submodule checkout with GitHub HTTP 500 errors, before the build reached this PR's code. I reran the failed inductor and torchtitan jobs.

**Comment by jansel** [comment]
My agent says this PR is being closed as part of a cleanup of open agentic PRs to focus review bandwidth on a smaller set; the linked issue remains open.
